include/private/action.h

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_PRIVATE_ACTION_H
+#define __LABWC_PRIVATE_ACTION_H
+
+/* Don't include ourself as search path starts at current directory */
+#include "../action.h"
+
+enum action_arg_type {
+	LAB_ACTION_ARG_STR = 0,
+	LAB_ACTION_ARG_INT,
+	LAB_ACTION_ARG_DOUBLE,
+	LAB_ACTION_ARG_BOOL,
+	LAB_ACTION_ARG_BOX,
+};
+
+struct action_arg_str {
+	struct action_arg base;
+	char *value;
+};
+
+struct action_arg_int {
+	struct action_arg base;
+	int value;
+};
+
+struct action_arg_double {
+	struct action_arg base;
+	double value;
+};
+
+struct action_arg_bool {
+	struct action_arg base;
+	bool value;
+};
+
+struct action_arg_box {
+	struct action_arg base;
+	struct wlr_box value;
+};
```

Comment:
Only `struct action_arg_str` is used in this PR, the others are examples of how this action_arg framework could be used in the future